ratelimiter/ratelimiter.go

```diff
@@ -0,0 +1,76 @@
+package ratelimiter
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/http-proxy/filters"
+	"github.com/hashicorp/golang-lru"
+)
+
+var log = golog.LoggerFor("ratelimiter")
+
+type ratelimiter struct {
+	hostPeriods          map[string]time.Duration
+	hostAccessesByClient *lru.Cache
+	mx                   sync.Mutex
+}
+
+// New creates a new rate limiting filter that only allows access to the hosts
```

Comment:
The name `ratelimiter` is misleading because it limits not only the request rate, but also the hosts to visit. It can't serve as a general ratelimiter.

It's probably too late to rename it because http-proxy-lantern already used it, but maybe worth a package comment?